solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -111,7 +115,24 @@ pub mod solana_ibc {
     /// TODO(mina86): At the moment we’re operating on pretend tokens and each
     /// validator can set whatever stake they want.  This is purely for testing
     /// and not intended for production use.
+    ///
+    /// Can only be called through CPI from our staking program
+    /// (Program ID: 4EgHMraeMbgQsKyx7sG81ovudTkYN3XcSHpYAJayxCEG)
```

Comment:
I’d consider adding the program id to ChainData so that it doesn’t have to be hard-coded.